src/components/common/TokenAmountInput/index.tsx

```diff
@@ -67,6 +67,10 @@ const TokenAmountInput = ({
           {...register(TokenAmountFields.amount, {
             required: true,
             validate: validate ?? validateAmount,
+            onChange: (event) => {
+              event.target.value = event.target.value.trim()
+              return event
+            },
```

Comment:
As an additional thing, should we also remove leading zeroes?

Reply:
Great idea! I've added the logic to the `NumberField` directly.